libvast/src/system/explore_command.cpp

```diff
@@ -33,12 +34,22 @@
 using namespace caf;
 using namespace std::chrono_literals;
 
+namespace vast::detail {
+
+template <concepts::unsigned_integral T>
+auto non_overflowing_multiply(T lhs, T rhs) {
```

Comment:
Some nitpicking:

At least in rust this is called a `saturating_mul` (and I think I've heard the term elsewhere as well)

Also, you could save the division by using `__builtin_mul_overflow()` here.

Since it's already in `vast::detail` the intention probably was to pull it into a separate `detail/` header?